CMakeLists.txt

```diff
@@ -61,6 +61,10 @@ endif()
 # PIC (Position Independent Code) ensures .a files can be linked to executables that have PIE enabled
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+if (DO_DISABLE_EXCEPTIONS)
```

Comment:
>DO_DISABLE_EXCEPTIONS [](http://example.com/codeflow?start=4&length=21)

How about we go with opt-in for exceptions? DO_ENABLE_EXCEPTIONS. WIL does the same.

The disable word introduces a double negative when writing !DO_DISABLE_EXCEPTIONS which I'd like to avoid.

Reply:
Yeah I had considered this but the problem with opt-in is that it doesn't remove the throwing code.
There is internal code which may still utilize try/catch or throw keywords that need to be disabled by an application compiling with -fno-except

Reply:
I don't follow. Our code should not use try/catch unless DO_ENABLE_EXCEPTIONS is defined. If it is about 3rd party libs, then it applies to DO_DISABLE_EXCEPTIONS as well.

Reply:
We can do that, but we'd need to bump the major version of the SDK after doing so, because a consumer would now need to set that compile option. With the current implementation, we don't necessarily hae to bump the majorversion since it's only opt-in to disable

Reply:
I'm going to leave it as is for now, and I can check in a subsequent change using DO_ENABLE_EXCEPTIONS and bumping the majorversion when ready

Reply:
Note that only API changes require major version increment, not changes in the way it builds. Although one could argue that adding noexcept is a change to the API.

We are in preview phase so we should take advantage of it for such changes (thinking about DU here). Moreover this is the first time Edge will consume this, so let's make this change now.

Reply:
Yes, will do. I had mentioned that after splitting this PR, we can easily make the exception change and choose to increment the majorversion (or leave as is). I'll make the change in a subsequent PR